### ISSUE TITLE
Policy on i18n

### ISSUE THREAD

**Issue by milovanderlinden** [openSUSE/osem#688]
This is a question. What is the policy on i18n. Will OSEM only use live translate for the public facing pages? Or is it also possible to do a full translation, also for the backend? In a lot of conferences, the people organizing are not technical (and often not english) for them a i18n in the backend might be good.

Any thoughts?

**Comment by hennevogel** [comment]
We use transifex for everything :-)

**Comment by milovanderlinden** [comment]
Well, not really I guess? Because nothing is translating to dutch. Could you shed some light on the how?

**Comment by ChrisBr** [comment]
@milovanderlinden We use http://www.transifex.com/ for a live translation. If you want a dutch support you have to do the translation on transifex. I guess currently we support only english and greek (@differentreality right?).

@Doctor-Who is also currently working on a translation if I'm right (but I guess french?)

For more information @differentreality is the right person to get in touch because she implemented the whole Transifex / translation stuff.

**Comment by differentreality** [comment]
@milovanderlinden I don't see any translations in Dutch in trasifex...
Once you make some translations, we need to publish the translations.
After they are published, then we get to see them in the app.

Does this cover your question? Or are you having other kinds of trouble translating?

Short description of the process:
https://github.com/openSUSE/osem/wiki/Translation-with-Transifex

**Comment by milovanderlinden** [comment]
Thanks for answering! I followed the links in the process description, but they end up dead. This one:

https://www.transifex.com/projects/p/osem/

works. I signed up for the team.

**Comment by milovanderlinden** [comment]
I am currently in the process of translating purely on the backend with i18n. I would like to know if this would be something that might be of use to the general project. If so, I would like to create a pull request in the near future.

The way I translate is by using:

    = t('some.key', default: 'Some key value that was the original untranslated text'

By using this method, the system corresponds as it used to, even with the live transifex translation. Would this be an enhancement or do you guys say: "Nah.. we don't want that"

**Comment by hennevogel** [comment]
I would definitely want to incorporate this into some translation framework like transifex, weblate or similar where translators hang out. Otherwise this will just be files that bitrot in our repository....

**Comment by milovanderlinden** [comment]
I completely agree! That is also why I am using the default: method. This way it should be possible to easily filter all strings that can be translated. I read through the transifex documentation, they have a way of setting up .yml files for translation, so that shouldn't be to hard. Currently I am using a single file per translation; where en.yml is as good as empty because this will always fallback to default. I will investigate transifex integration and will try to come up with a valuable, all in solution.

I am currently investigating https://github.com/crewmate/crewmate/issues/136 seeing what I can learn from their approach.

**Comment by milovanderlinden** [comment]
I set up a test project at https://www.transifex.com/organization/osem_test/dashboard/osem_backend this project is integratedd in my fork with a file in .tx/config that one can initiate with

    tx pull

I am now looking at how this can be integrated with rails and possibly a rake task.

**Comment by milovanderlinden** [comment]
As I personally have bad experiences with "monster" pulls, I will not update my fork yet but will create a small test-pull that can be reviewed by you guys. If it is a workable solution, I will slowely move foreward keeping you guys in the lead.
